libsolidity/codegen/CompilerUtils.cpp

```diff
@@ -656,6 +656,15 @@ void CompilerUtils::convertType(Type const& _typeOnStack, Type const& _targetTyp
 		solAssert(_typeOnStack == _targetType, "Invalid type conversion requested.");
 		break;
 	}
+
+	// Check the conversion result fits in a range.
+	if (targetTypeCategory == Type::Category::Enum)
+	{
+		EnumType const* enumType = dynamic_cast<decltype(enumType)>(&_targetType);
+		solAssert(enumType, "targetTypeCategory is wrong.");
+		m_context << u256(enumType->numberOfMembers()) << Instruction::DUP2 << Instruction::LT << Instruction::ISZERO;
```

Comment:
If we fix that condition, the assembly can be simplified. Either we do that in this PR or separately. However, since limiting enum values is a breaking change in some way  (will break code) and that construct is useless after merging this PR we could as well just fix it here.